back/app/services/create_page_service.rb

```
@@ -0,0 +1,41 @@
+class CreatePageService
+  def initialize(page, user)
+    @page = page
+    @user = user
+  end
+
+  def call
+    ActiveRecord::Base.transaction { create_page }
+  end
+
+  private
+
+  attr_reader :page, :user
+
+  def create_page
+    prepare_navbar_items(page.navbar_item)
+
+    result = page.save
+    return result unless result
+
+    page.update(body_multiloc: TextImageService.new.swap_data_images(page, :body_multiloc))
```

I would prefer `update!` to ensure this doesn't silently fail. Could we still consider calling the sidefx service from within this service? It's not a big deal if you prefer not, but I would find it useful to keep the overview of this kind of logic in the sidefx service.

---

Ok. I'll do it

---

Done